runtime/commonMain/src/kotlinx/serialization/modules/PolymorphicModuleBuilder.kt

```diff
@@ -59,6 +61,21 @@ public class PolymorphicModuleBuilder<Base : Any> internal constructor(
     @ImplicitReflectionSerializer
     public inline fun <reified T : Base> subclass(): Unit = addSubclass(T::class, serializer())
 
+    /**
+     * Registers serializer provider that will be invoked if no polymorphic serializer is present.
+     * [defaultSerializerProvider] can be stateful and lookup a serializer for the missing type dynamically.
+     *
+     * Typically, if the class is not registered in advance, it is not possible to know the structure of the unknown
+     * type and have a precise serializer, so the default serializer has limited capabilities.
+     * To have a structural access to the unknown data, it is recommended to use [JsonTransformingSerializer] subclass.
+     */
+    public fun default(defaultSerializerProvider: (className: String) -> DeserializationStrategy<out Base>?) {
```

Comment:
I've decided not to mention that it is only applicable for deserialization (it should be more or less obvious?) and express it in the return type instead